core/src/main/scala/akka/kafka/internal/TransactionalProducerStage.scala

```diff
@@ -99,18 +100,22 @@ private object TransactionalProducerStage {
  * Transaction (Exactly-Once) Producer State Logic
  */
 private final class TransactionalProducerStageLogic[K, V, P](stage: TransactionalProducerStage[K, V, P],
-                                                             producer: Producer[K, V],
+                                                             producerProvider: String => Producer[K, V],
```

Comment:
We'll get conflicts with the Akka Discovery PR here...

Reply:
We sure did!  It's been handled now.